src/main/scala/cromwell/binding/values/WdlObject.scala

```diff
@@ -1,14 +1,41 @@
 package cromwell.binding.values
 
 import cromwell.binding.Call
-import cromwell.binding.types.{WdlCallOutputsObjectType, WdlObjectType}
+import cromwell.binding.types._
+import cromwell.util.{TryUtil, FileUtil}
+
+import scala.util.{Failure, Success, Try}
 
 trait WdlObjectLike {
   def value: Map[String, WdlValue]
 }
 
+object WdlObject {
+
+  def coerceObject(m: Map[String, String]): WdlObject = {
+    val coerced = WdlMap.coerceMap(m, WdlMapType(WdlStringType, WdlStringType)).value map {
+      case (k,v) => k.asInstanceOf[WdlString].value -> v
+    }
+    WdlObject(coerced)
+  }
+
+  def fromTsv(tsv: String): Try[WdlObject] = {
+    FileUtil.parseTsv(tsv) match {
+      case Success(table) if table.isEmpty => Success(WdlObject(Map.empty[String, WdlValue]))
+      case Success(table) if table.length != 2 => Failure(new UnsupportedOperationException("TSV must be 2 rows to convert to an Object"))
+      case Success(table) => Try(coerceObject((table.head zip table.last).toMap))
+      case Failure(e) => Failure(e)
+    }
+  }
+
+}
+
 case class WdlObject(value: Map[String, WdlValue]) extends WdlValue with WdlObjectLike {
   val wdlType = WdlObjectType
+
+  override def toWdlString: String =
+    "object {" + value.map {case (k,v) => s"$k: ${v.toWdlString}"}.mkString(", ") + "}"
```

Comment:
https://groups.google.com/forum/#!topic/workflow-description-language/h5W7MVkoeGk for discussion about the object literal syntax.  `object {...}` is a bit hacked into the parser (from back when I was writing PyWDL and I just needed something quick)

I don't think it needs to be addressed in this PR though.